unicorn/src/repository/zookeeper.cpp

```diff
@@ -0,0 +1,124 @@
+#include "zookeeper.hpp"
+
+#include <cocaine/context.hpp>
+#include <cocaine/logging.hpp>
+
+static int do_log(void* handler, const char* buf, size_t size) {
+    cocaine::logging::logger_t* log = static_cast<cocaine::logging::logger_t*>(handler);
+
+    std::string message(buf, size);
+    while (message.size() > 0 && message.back() == '\n') {
+        message.resize(message.size() - 1);
+    }
+
+    auto pos = message.find("ZOO_");
+    if (pos == std::string::npos || pos + 4 >= message.size()) {
+        return 0;
+    }
+
+    int severity = 0;
+    switch (message[pos + 4]) {
+    case 'D':
+        severity = 0;
+        break;
+    case 'I':
+        severity = 1;
+        break;
+    case 'W':
+        severity = 2;
+        break;
+    case 'E':
+        severity = 3;
+        break;
+    default:
+        ;
+    }
+
+    pos = message.find(": ");
```

Comment:
It's better to look for this colon after prefix just in case(and it will be more efficient).